src/site_upload/cache_api/cache_api.py

```diff
@@ -43,17 +44,23 @@ def cache_api_data(s3_client, s3_bucket_name: str, db: str, target: str) -> None
         if dp_ids is None:  # pragma: no cover
             continue
         for dp_id in dp_ids:
-            if dp_id != dp:
+            if dp_id not in dp:
                 continue
-            version = dp_id.split("__")[2]
+            if "__" in dp_id:
+                version = dp_id.split("__")[2]
+            else:
+                version = dp_id
             dp_dict = {
                 **dp_detail,
                 **dp_ids[dp_id],
                 "version": version,
                 "id": f"{dp_detail['study']}__{dp_detail['name']}__{version}",
             }
-            if "__flat" in dp:
+            if "__flat" in dp_dict["s3_path"]:
+                site = dp_dict["s3_path"].split("/")[5]
```

Comment:
More magic path numbering 😕

Reply:
there's an issue about it - #143 - I just don't have the time right now with the backlog of other things i've got